repository_service_tuf/cli/artifact/repository.py

```diff
@@ -0,0 +1,261 @@
+# SPDX-FileCopyrightText: 2023 VMware Inc
+#
+# SPDX-License-Identifier: MIT
+
+
+import base64
+
+from click import Context
+from dynaconf.loaders.yaml_loader import write  # type: ignore
+
+from repository_service_tuf.cli import click, console
+from repository_service_tuf.cli.artifact import artifact
+
+
+def write_config(settings_path, settings_data, merge=True):
+    write(settings_path, settings_data, merge)  # pragma: no cover
+
+
+@artifact.group()
+@click.pass_context
+def repository(context) -> None:
+    """
+    Repository management.
+    """
+
+
+@repository.command()
+@click.option(
+    "-a",
+    "--all",
+    help="Show all the configured repositories.",
+    is_flag=True,
+    required=None,
+)
+@click.pass_context
+@click.argument("repository", required=False)
+def show(
+    context: Context,
+    all: bool,
+    repository,
+) -> None:
+    """
+    List configured repositories.
+    """
+
+    rstuf_config = context.obj.get("settings")
+
+    if all:
+        if rstuf_config.get("REPOSITORIES"):
+            for repo in rstuf_config.get("REPOSITORIES"):
+                console.print(repo)
+        else:
+            console.print("There are no configured repositories")
+            return
+
+    if repository:
+        if rstuf_config.get("REPOSITORIES") and rstuf_config[
+            "REPOSITORIES"
+        ].get(repository):
+            console.print(rstuf_config["REPOSITORIES"].get(repository))
+        else:
+            console.print(
+                f"Repository {repository} is missing in your configuration"
+            )
```

Comment:
I think if the repository is empty we can also show the current repository details.

```suggestion
    if all:
        if rstuf_config.get("REPOSITORIES"):
            for repo in rstuf_config.get("REPOSITORIES"):
                console.print(repo)
        else:
            raise click.ClickException("There are no configured repositories")

    elif repository:
        if rstuf_config.get("REPOSITORIES") and rstuf_config[
            "REPOSITORIES"
        ].get(repository):
            console.print(rstuf_config["REPOSITORIES"].get(repository))
        else:
            raise click.ClickException(
                f"Repository {repository} is missing in your configuration"
            )
    else:
        console.print(
            rstuf_config["REPOSITORIES"].get(
                rstuf_config["CURRENT_REPOSITORY"])
        )
            
```